File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        def isdiv(s,S):
            i=0
            while(i<len(S)):
                if s[i%len(s)]!=S[i]:
                    return False
                i+=1
            i-=1
            if i%len(s)==len(s)-1:
                return True
            return False
                
        if str1==str2:
            return str1
        if str1[0]!=str2[0]:
            return ''
        ans=''
        temp=''
        for i in range(min(len(str1),len(str2))):
            if str1[i]!=str2[i]:
                return ans
            temp+=str1[i]
            if isdiv(temp,str1) and isdiv(temp,str2):
                ans=temp
        return ans
            
                
        
        

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        string div="";
        string check="";
        int n=min(str1.size(),str2.size());
        for(int i=0;i<n;i++)
        {
            check+=str1[i];
            if(divide(str1,check) && divide(str2,check))div=check;
        }
        return div;
        
        
    }
private:
    bool divide(string bada,string chota)
    {
        int j=0;
        for(int i=0;i<bada.size();i++)
        {
            if(bada[i]!=chota[j]){
                return 0;}
            else{
                j=(j+1)%chota.size();
            }   
        }
        if(j==0)
            return 1;
        return 0;
    }
};